carbons/carbons_test.go

```diff
@@ -104,3 +105,63 @@ func TestWrapSent(t *testing.T) {
 		t.Fatalf("wrong output:\nwant=%s,\n got=%s", expected, out)
 	}
 }
+
+var unwrapTestCases = [...]struct {
+	unwrappedXML string
+	reason       string
```

Comment:
We can get rid of the reason and checking that the delay got unmarshaled here, I think. We're already testing that `forward.Unwrap` works and this relies on that, so I don't think we need to duplicate those tests. Just that the output XML is what we expect and the value of sent/received is correct should cover all our bases. Eventually we should probably add some tests that check that an error gets returned in various cases too (namespace wrong, sent/received name wrong, etc.)

Reply:
Humm, indeed these tests are a little too cautious (just in case the function does not use forward.Unwrap correctly or not at all). I will get rid of this redundancy and add more error related cases

Reply:
Added also some error related tests for forward.Unwrap (https://github.com/mellium/xmpp/pull/214)